lib/src/main/java/com/thenewboston/data/source/bankapi/ValidatorDataSource.kt

```diff
@@ -0,0 +1,35 @@
+package com.thenewboston.data.source.bankapi
+
+import com.thenewboston.common.http.NetworkClient
+import com.thenewboston.common.http.Outcome
+import com.thenewboston.data.dto.bankapi.validatordto.ValidatorDTO
+import com.thenewboston.data.dto.bankapi.validatordto.ValidatorListDTO
+import io.ktor.client.request.get
+import io.ktor.util.KtorExperimentalAPI
+
+private const val VALIDATORS_ENDPOINT = "/validators"
+
+private const val GENERIC_ERROR_MESSAGE = "An error occurred"
+
+@KtorExperimentalAPI
+internal class ValidatorDataSource(private val networkClient: NetworkClient) {
+
+    suspend fun fetchValidators(): Outcome<ValidatorListDTO> = try {
+        val validators = networkClient.client.get<ValidatorListDTO>(VALIDATORS_ENDPOINT)
+
+        when {
+            validators.results.isNullOrEmpty() -> Outcome.Error("Could not fetch validators", null)
+            else -> Outcome.Success(validators)
+        }
+    } catch (e: Exception) {
+        Outcome.Error(GENERIC_ERROR_MESSAGE, e)
+    }
+
+    suspend fun fetchValidator(nodeIdentifier: String): Outcome<ValidatorDTO> = try {
+        val urlSuffix = "$VALIDATORS_ENDPOINT/$nodeIdentifier"
+        val validator = networkClient.client.get<ValidatorDTO>(urlSuffix)
+        Outcome.Success(validator)
+    } catch (e: Exception) {
+        Outcome.Error(GENERIC_ERROR_MESSAGE, e)
+    }
```

Comment:
Let's try create a reusable function for making the api calls instead of try/catchs everywhere.

An example, we can create this in a class in http package as NetworkUtils.kt
```
suspend fun <T : Any> makeApiCall(call: suspend () -> Result<T>, errorMessage: String): Outcome<T> {
    return try {
        call()
    } catch (e: Exception) {
        // An exception was thrown when calling the API so we're converting this to an IOException
        Result.Error(IOException(errorMessage, e))
    }
}
```

Example usage:
```
suspend fun validators(): Outcome<ValidatorListDTO> {
        val validators = networkClient.client.get<ValidatorListDTO>(VALIDATORS_ENDPOINT)

        when {
            validators.results.isNullOrEmpty() -> Outcome.Error("Could not fetch validators", null)
            else -> Outcome.Success(validators)
        }
}
```
Then wrap our call like this:
```
    suspend fun fetchValidators() = makeApiCall(
        call = { fetchValidators() },
        errorMessage = "Failed to retrieve banks"
    )
```

Reply:
Good point, the try/catchs were annoying me as well, and I was also wondering if I should create an `IOException`.